src/matching/basin_enclosure.jl

```diff
@@ -104,11 +85,13 @@ function matching_map(
         end
     end
     # next up are the co-flowing attractors
-    grouped_flows = _grouped_flows(flow)
+    allnewids = keys(current_attractors) #needed because flow only includes new ids that prev attractors flowed onto
+    grouped_flows = _grouped_flows(flow, allnewids) #map all current ids to prev ids that flowed onto them (if the att is new, no prev ids flowed, and the corresponding entry is empty)
+
     # notice the keys of `grouped_flows` are new IDs, same as with `rmap`.
     for (new_ID, old_flowed_to_same) in grouped_flows
-        if length(old_flowed_to_same) == 0
-            continue # none of the old IDs converged to the current `new_ID`
+        if length(old_flowed_to_same) == 0 # none of the old IDs converged to the current `new_ID`
+            rmap[new_ID] = next_id #necessary to make rmap complete and avoid skipping keys
```

Comment:
This is not correct because `next_id` is not incremented. Imagine that in `grouped_flows` there is more than one element `old_flowed_to_same` that has length > 0. Then, all `new_ID` will get the same `next_id`. What you have to do is add a new line after `rmap[new_ID] = next_id` that simply does `next_id += 1`.

Reply:
true!

Reply:
true!